### PR TITLE
feat: Shortcodes UI implementation

### PR DESCRIPTION
**Pull request by nivedin**
### Description
This PR introduces a feature in which authenticated user can view their created shortcodes and can delete or copy them.

### Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-app/helpers/shortcodes/ShortCodeListAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        this.timeoutHandle = setTimeout(() => func(), POLL_DURATION)
+      }
+    }
+
+    func()
+  }
+
+  public dispose() {
+    this.isDispose = true
+    clearTimeout(this.timeoutHandle as any)
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/helpers/shortcodes/ShortCodeListAdapter.ts:49]: TypeScript is raising a valid point here, what if there is no timeout handle (for e.g, already disposed or when not initialized). Should we call `clearTimeout` then ?

File: packages/hoppscotch-app/pages/profile.vue [hunk 2]
```diff
@@ -273,6 +355,39 @@ const sendEmailVerification = () => {
...
+const loading = computed(
+  () => adapterLoading.value && myShortCodes.value.length === 0
+)
+
+const deleteShortCode = (codeID: string) => {
+  pipe(
+    backendDeleteShortcode(codeID),
+    TE.match(
+      (err) => {
+        toast.error(`${t("error.something_went_wrong")}`)
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/pages/profile.vue:376]: We can detect exactly what happened here. For example, if its a network error (see err.type)

File: packages/hoppscotch-app/pages/profile.vue [hunk 3]
```diff
@@ -273,6 +355,39 @@ const sendEmailVerification = () => {
...
+        console.error(err)
+      },
+      () => {
+        toast.success(`${t("shortcodes.deleted")}`)
+        refetchShortcodes()
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/pages/profile.vue:381]: Do we have to refetch the entire list again just for deletion ? The adapter should handle deletion without refetching everything.

File: packages/hoppscotch-app/pages/profile.vue [hunk 4]
```diff
@@ -273,6 +355,39 @@ const sendEmailVerification = () => {
...
+      }
+    )
+  )()
+}
+
+const refetchShortcodes = () => {
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/pages/profile.vue:387]: This function doesn't need to exist.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
Please look into the given comments.

Also, please maintain a single naming convention across the codebase, for example, the backend operations define it as a 'Shortcode', not a 'ShortCode' or 'Short Code'. Different naming styles are being used throughout the code here, normalize it into a single style.

(it should be Shortcode/shortcode, not shortCode/short-code/Shortcode etc.)

**Comment by liyasthomas** [comment]
@nivedin this branch seems broken. Please rebase to the latest `main` and update the PR.

**Pull request closed without merging**